module/VuFind/src/VuFind/Auth/Shibboleth.php

```diff
@@ -44,255 +42,23 @@
-        $catPassword = null;
-
-        // Has the user configured attributes to use for populating the user table?
-        $attribsToCheck = [
-            'cat_username', 'cat_password', 'email', 'lastname', 'firstname',
-            'college', 'major', 'home_library'
-        ];
-        foreach ($attribsToCheck as $attribute) {
-            if (isset($shib->$attribute)) {
-                $value = $request->getServer()->get($shib->$attribute);
-                if ($attribute == 'email') {
-                    $user->updateEmail($value);
-                } elseif ($attribute != 'cat_password') {
-                    $user->$attribute = ($value === null) ? '' : $value;
-                } else {
-                    $catPassword = $value;
-                }
-            }
-        }
-
-        // Save credentials if applicable. Note that we want to allow empty
-        // passwords (see https://github.com/vufind-org/vufind/pull/532), but
-        // we also want to be careful not to replace a non-blank password with a
-        // blank one in case the auth mechanism fails to provide a password on
-        // an occasion after the user has manually stored one. (For discussion,
-        // see https://github.com/vufind-org/vufind/pull/612). Note that in the
-        // (unlikely) scenario that a password can actually change from non-blank
-        // to blank, additional work may need to be done here.
-        if (!empty($user->cat_username)) {
-            $user->saveCredentials(
-                $user->cat_username,
-                empty($catPassword) ? $user->getCatPassword() : $catPassword
-            );
-        }
-
-        // Add session id mapping to external_session table for single logout support
-        if (isset($shib->session_id)) {
-            $shibSessionId = $request->getServer()->get($shib->session_id);
-            if (null !== $shibSessionId) {
-                $localSessionId = $this->sessionManager->getId();
-                $externalSession = $this->getDbTableManager()
-                    ->get('ExternalSession');
-                $externalSession->addSessionMapping(
-                    $localSessionId, $shibSessionId
-                );
-                $this->debug(
-                    "Cached Shibboleth session id '$shibSessionId' for local session"
-                    . " '$localSessionId'"
-                );
-            }
-        }
-
-        // Save and return the user object:
-        $user->save();
-        return $user;
-    }
-
-    /**
-     * Get the URL to establish a session (needed when the internal VuFind login
-     * form is inadequate).  Returns false when no session initiator is needed.
-     *
-     * @param string $target Full URL where external authentication method should
-     * send user after login (some drivers may override this).
-     *
-     * @return bool|string
+     * @throws \VuFind\Exception\Auth
+     * @return array Shibboleth configuration to use
```

Comment:
We should add:

<pre>
     *
     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
</pre>

to the bottom of this comment block to suppress warnings about the intentionally unused formal parameter here.